### ISSUE TITLE
Any plans to implement sqlite3_update_hook?

### ISSUE THREAD

**Issue by reinhardbluelab** [simolus3/sqlite3.dart#61]
We have a trigger (before insert) that inserts into another table. Using the watch() function in Drift does not seem to fire the callback for the update of the other table; it does fire the update for the original (i.e. Drift initiated) table though.

https://www.sqlite.org/c3ref/update_hook.html

Any thoughts on implementing this so that a Dart callback could be called for all (and not only Drift insertion) table updates?

Keep up the great work!

**Comment by simolus3** [comment]
> Using the watch() function in Drift does not seem to fire the callback for the update of the other table

It definitely should though! Are you defining the trigger in a `.drift` file or are you creating it manually? For triggers that Drift knows about, stream updates should work.

I think update hooks could be a useful addition to this package regardless of Drift though.

**Comment by reinhardbluelab** [comment]
At the moment we're creating these triggers manually since they rely on a complex view that doesn't work in a `.drift` file. We're just running these in a _customStatement_.

**Comment by simolus3** [comment]
I have implemented update hooks in 8d308c15677914aa5a637249514a9cfe57927438.

> At the moment we're creating these triggers manually since they rely on a complex view that doesn't work in a `.drift` file

If drift crashes or provides invalid analysis for a valid sqlite3 view, I'd love to take a look at that. If you're using `customStatement`s to create triggers, you need to tell drift how these affect stream updates manually (the generator should otherwise be able to figure it out).

```dart
  @override
  StreamQueryUpdateRules get streamUpdateRules {
    return StreamQueryUpdateRules(
      [
        ...super.streamUpdateRules.rules,
        WritePropagation(
          on: TableUpdateQuery.onTable(tableThatTheTriggerIsListeningOn),
          result: [
            TableUpdate.onTable(tableThatMightBeWrittenInto),
          ],
        )
      ],
    );
  }
```
